crates/voicevox_core/src/engine/kana_parser.rs

```diff
@@ -0,0 +1,280 @@
+                    accent_phrase.pause_mora = Some(MoraModel {
+                        text: PAUSE_DELIMITER.to_string(),
+                        consonant: None,
+                        consonant_length: None,
+                        vowel: "pau".to_string(),
+                        vowel_length: 0.0,
+                        pitch: 0.0,
+                    });
+                }
+                accent_phrase.is_interrogative = is_interrogative;
+                accent_phrase
+            };
+            parsed_result.push(accent_phrase);
+            phrase.clear();
+        } else {
+            phrase.push(letter);
+        }
+    }
+    Ok(parsed_result)
+}
+
+#[allow(dead_code)] // TODO: remove this feature
+fn create_kana(accent_phrases: &[AccentPhraseModel]) -> String {
+    let mut text = String::new();
+    for phrase in accent_phrases {
+        let moras = &phrase.moras;
+        for (index, mora) in moras.iter().enumerate() {
+            if ["A", "E", "I", "O", "U"].contains(&mora.vowel.as_ref()) {
+                text.push(UNVOICE_SYMBOL);
+            }
+            text.push_str(&mora.text);
+            if index + 1 == phrase.accent {
+                text.push(ACCENT_SYMBOL);
+            }
+        }
+        if phrase.is_interrogative {
+            text.push(WIDE_INTERROGATION_MARK);
+        }
+        text.push(if phrase.pause_mora.is_some() {
+            PAUSE_DELIMITER
+        } else {
+            NOPAUSE_DELIMITER
+        });
+    }
+    text.pop(); // remove last delimiter
+    text
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::engine::mora_list::MORA_LIST_MINIMUM;
+
+    #[test]
+    fn test_text2mora_with_unvoice() {
+        let text2mora = &TEXT2MORA_WITH_UNVOICE;
+        assert_eq!(text2mora.len(), MORA_LIST_MINIMUM.len() * 2 - 2); // added twice except ン and ッ
+        let lis = [
+            (Some("da"), "ダ"),
+            (Some("N"), "ン"),
+            (Some("cl"), "ッ"),
+            (Some("sho"), "ショ"),
+            (Some("u"), "ウ"),
+            (Some("gA"), "_ガ"),
+            (Some("byO"), "_ビョ"),
+            (Some("O"), "_オ"),
+            (None, "fail"),
+        ];
+        for (mora, text) in lis {
+            let res = text2mora.get(text);
+            assert_eq!(mora.is_some(), res.is_some());
+            if let Some(res) = res {
+                let mut m = String::new();
+                if let Some(ref c) = res.consonant {
+                    m.push_str(c);
+                }
+                m.push_str(&res.vowel);
+                assert_eq!(m, mora.unwrap());
+            }
+        }
+    }
+
+    #[test]
+    fn test_text_to_accent_phrase() {
+        let text_ok = ["ア_シタ'ワ", "ユウヒガ'", "_キ'レイ"];
+        let text_err = [
+            "アクセントナシ",
+            "アクセ'ント'タクサン'",
+            "'アクセントハジマリ",
+            "不明な'文字",
+        ];
+        for text in text_ok {
+            assert!(text_to_accent_phrase(text).is_ok());
```

Comment:
`assert!` を使ったアサーションは、期待と異なる出力だった時にその出力を表示させるのが良いと思います……！（他の同様の箇所についてもよろしくお願いします）

```suggestion
            let result = text_to_accent_phrase(text);
            assert!(result.is_ok(), "{:?}", result);
```